### PR TITLE
EL-1538: Add Django Jinja2 templates

### PR DESCRIPTION
**Pull request by MazOneTwoOne**
## What does this pull request do?

- working example of using jinja2 template package (`govuk-frontend-jinja`)
- removing commented out code in 'MIDDLEWARE_CLASSES' & 'TEMPLATES'
- comments added in for future devs
- new `jinja2.py` file for jinja2 environments

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

- [ ] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: requirements/generated/requirements-dev.txt [hunk 1]
```diff
@@ -24,7 +24,7 @@ click==8.1.7
     # via pip-tools
 distlib==0.3.8
     # via virtualenv
-django==5.0.6
+django==4.2.13
```
Comment (exonian) [inline requirements/generated/requirements-dev.txt:27]: Did we mean to downgrade this...?
Perhaps our pinning in the .in file needs to specify at least 5 as well as less than 5.1?

File: requirements/source/requirements-dev.in [hunk 2]
```diff
@@ -4,4 +4,5 @@
 pip-tools==7.4.1
 beautifulsoup4<5
 pytest-playwright~=0.5.0
+django-jinja==2.10.2
```
Comment (exonian) [inline requirements/source/requirements-dev.in:7]: This is already in the base requirements, which this file extends

Comment (MazOneTwoOne) [reply requirements/source/requirements-dev.in:7]: oh yes, now removed.

File: requirements/generated/requirements-dev.txt (hunk 1 above)
Comment (MazOneTwoOne) [reply requirements/generated/requirements-dev.txt]: nope - now resolved

**Pull request merged**